### PR TITLE
Fix issues with lines being hidden in DSG mode

### PR DESCRIPTION
**Pull request by randallfrank**
- lines can now be displayed from a DSG source
- inline GLB textures are now supported
- add kind notations to objects in the tree
- remove support for older USD APIs
- make end caps rounded

### CODE REVIEW

File: src/ansys/pyensight/core/utils/omniverse_dsg_server.py [hunk 1]
```diff
@@ -299,16 +299,10 @@ def create_dsg_mesh_block(
             mesh.CreateFaceVertexCountsAttr([3] * (conn.size // 3))
             mesh.CreateFaceVertexIndicesAttr(conn)
             if (tcoords is not None) and variable:
-                # USD 22.08 changed the primvar API
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/omniverse_dsg_server.py:302]: The version of the Usd module is controlled by the wheel dependencies, so we can remove backward compatibility.

File: src/ansys/pyensight/core/utils/omniverse_dsg_server.py [hunk 2]
```diff
@@ -372,7 +366,7 @@ def create_dsg_lines(
     ):
         # 1D texture map for variables https://graphics.pixar.com/usd/release/tut_simple_shading.html
         # create the part usd object
-        partname = self.clean_name(name + part_hash.hexdigest())
+        partname = self.clean_name(name + part_hash.hexdigest()) + "_l"
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/omniverse_dsg_server.py:369]: This was the critical issue.  The name of the object needed to be different from the triangles object to make sure this gets generated at all.

File: src/ansys/pyensight/core/utils/omniverse_dsg_server.py [hunk 3]
```diff
@@ -407,21 +399,22 @@ def create_dsg_lines(
             lines.CreateTypeAttr().Set("linear")
             lines.CreateWidthsAttr([width])
             lines.SetWidthsInterpolation("constant")
+            # Rounded endpoint are a primvar
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/omniverse_dsg_server.py:402]: Make the default to be "rounded" end caps.

File: src/ansys/pyensight/core/utils/omniverse_dsg_server.py [hunk 4]
```diff
@@ -650,10 +642,17 @@ def create_dsg_group(
...
             group_prim = UsdGeom.Xform.Define(self._stage, path)
             # At present, the group transforms have been cooked into the vertices so this is not needed
-            matrixOp = group_prim.AddXformOp(
+            matrix_op = group_prim.AddXformOp(
                 UsdGeom.XformOp.TypeTransform, UsdGeom.XformOp.PrecisionDouble
             )
-            matrixOp.Set(Gf.Matrix4d(*matrix).GetTranspose())
+            matrix_op.Set(Gf.Matrix4d(*matrix).GetTranspose())
+            # Map kinds
+            kind = Kind.Tokens.group
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/omniverse_dsg_server.py:650]: Add support for "kind" values on cases, parts and groups.

File: src/ansys/pyensight/core/utils/omniverse_glb_server.py [hunk 5]
```diff
@@ -359,7 +359,13 @@ def upload_file(self, glb_filename: str, timeline: List[float] = [0.0, 0.0]) ->
             # Walk texture nodes -> DSG Variable buffers
             for tex_idx, texture in enumerate(self._gltf.textures):
                 image = self._gltf.images[texture.source]
-                raw_png = self._gltf.get_data_from_buffer_uri(image.uri)
+                if image.uri is None:
```
Comment (randallfrank) [inline src/ansys/pyensight/core/utils/omniverse_glb_server.py:362]: For EnSight generated GLB files, the 1D texture can be inline with the data blob.  In this case, there is no image.url and the buffer view is used to get the PNG payload.

**Review by mariostieriansys — APPROVED** [review]

**Pull request merged**